stdlib/public/core/Sequence.swift

```diff
@@ -877,6 +877,13 @@ extension Sequence {
   public func filter(
     _ isIncluded: (Element) throws -> Bool
   ) rethrows -> [Element] {
+    return try _filter(isIncluded)
```

Comment:
This will mean ContiguousArray and ArraySlice will also return an Array from filter, not Self? Which is... probably the right thing? Worth checking my thinking though.